overlord/oddjobstate/oddjobmgr.go

```diff
@@ -0,0 +1,96 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package oddjobstate
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"gopkg.in/tomb.v2"
+
+	"github.com/snapcore/snapd/overlord/state"
+)
+
+// OddJobManager is responsible for running arbitrary commands as tasks.
+type OddJobManager struct {
+	runner *state.TaskRunner
+}
+
+// Manager returns a new OddJobManager.
+func Manager(st *state.State) *OddJobManager {
+	runner := state.NewTaskRunner(st)
+	runner.AddHandler("exec", doExec, nil)
```

Comment:
we don't have a lot of single verb handler names,  "exec-command" ?  "run-command" to parallel "run-hook" ?